### PR TITLE
schedule: RegionScore consider the compression ratio for each tikv

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

try to fix https://github.com/tikv/pd/issues/3544

### What is changed and how it works?
consider the compression ratio for eatch tikv

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
before:
image
after:
image

### Release note

- let `RegionScoreV2` consider the compression ratio for each tikv

### CODE REVIEW

File: server/schedulers/balance_test.go [hunk 1]
```diff
@@ -848,6 +848,31 @@ func (s *testBalanceRegionSchedulerSuite) TestStoreWeight(c *C) {
 	testutil.CheckTransferPeer(c, sb.Schedule(tc)[0], operator.OpKind(0), 1, 3)
 }
 
+func (s *testBalanceRegionSchedulerSuite) TestIssue3544_UsageRatio(c *C) {
```
Comment (rleungx) [inline server/schedulers/balance_test.go:851]: I think a better naming should be TestUsageRatio and add comment `# Issue 3544`?

File: pkg/mock/mockcluster/mockcluster.go [hunk 2]
```diff
@@ -406,13 +406,23 @@ func (mc *Cluster) UpdateStoreRegionSize(storeID uint64, size int64) {
...
+	newStats.UsedSize = uint64(store.GetRegionSize() * mb)
 	newStore := store.Clone(
 		core.SetStoreStats(newStats),
 		core.SetRegionSize(size),
 	)
 	mc.PutStore(newStore)
 }
 
+// UpdateStoreRegionSizeAlone only updates store region size.
+func (mc *Cluster) UpdateStoreRegionSizeAlone(storeID uint64, size int64) {
```
Comment (rleungx) [inline pkg/mock/mockcluster/mockcluster.go:418]: prefer to use `UpdateStoreRegionSize` here And change the other one

File: server/core/store.go [hunk 3]
```diff
@@ -302,14 +302,23 @@ func (s *StoreInfo) regionScoreV1(highSpaceRatio, lowSpaceRatio float64, delta i
...
+	used := float64(s.GetUsedSize()) / mb
+	if s.GetRegionSize() == 0 || used == 0 {
+		amplification = 1
+	} else {
+		// because of rocksdb compression, region size is larger than actual used size
+		amplification = float64(s.GetRegionSize()) / used
+	}
 
+	// As the delta, not directly use the used size.
+	R := float64(s.GetRegionSize()+delta) / amplification
```
Comment (lhy1024) [inline server/core/store.go:321]: we will replace used_size with region_size?

Comment (nolouch) [reply server/core/store.go:321]: Yes, `region_size` may not correspond to the `used_size`, we prefer to balance `used_size`. see:
image
image

In the test, the region_size be more balance but  the `available_ratio` is not.

Comment (nolouch) [reply server/core/store.go:321]: cc @disksing

Comment (lhy1024) [inline server/core/store.go:321]:
```suggestion
	U := float64(s.GetRegionSize()+delta) / amplification
```

File: pkg/mock/mockcluster/mockcluster.go (hunk 2 above)
Comment (nolouch) [reply pkg/mock/mockcluster/mockcluster.go:418]: What the name about the other one?

**Review by Yisaer — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Comment by lhy1024** [comment]
@rleungx @disksing PTAL

File: server/core/store_option.go [hunk 4]
```diff
@@ -152,6 +152,15 @@ func SetRegionSize(regionSize int64) StoreCreateOption {
 	}
 }
 
+// SetStoreUsage sets the store usage for the store.
+func SetStoreUsage(usage uint64) StoreCreateOption {
```
Comment (HunDunDM) [inline server/core/store_option.go:156]: How about `SetUsedSize`?

File: server/core/store.go [hunk 5]
```diff
@@ -302,24 +302,33 @@ func (s *StoreInfo) regionScoreV1(highSpaceRatio, lowSpaceRatio float64, delta i
...
 func (s *StoreInfo) regionScoreV2(delta int64, deviation int) float64 {
+	var amplification float64
 	A := float64(float64(s.GetAvgAvailable())-float64(deviation)*float64(s.GetAvailableDeviation())) / gb
 	C := float64(s.GetCapacity()) / gb
-	R := float64(s.GetRegionSize() + delta)
 	var (
 		K, M float64 = 1, 256 // Experience value to control the weight of the available influence on score
 		F    float64 = 20     // Experience value to prevent some nodes from running out of disk space prematurely.
 	)
+	used := float64(s.GetUsedSize()) / mb
```
Comment (HunDunDM) [inline server/core/store.go:312]: Duplicate with `Available`?

Comment (nolouch) [reply server/core/store.go:312]: No, `A !=  C-U`, the used size is the `datadir` size of the tikv, but availabel is the remaining space of the disk.

File: pkg/mock/mockcluster/mockcluster.go (hunk 2 above)
Comment (rleungx) [reply pkg/mock/mockcluster/mockcluster.go:418]: How about UpdateStoreSize

**Comment by nolouch** [comment]
This PR will cause the Redundant scheduling, close it:
image

**Pull request closed without merging**